armory/utils/version.py

```diff
@@ -28,42 +29,58 @@
 from armory.logs import log
 
 
+class VersionError(Exception):
+    pass
+
+
 def to_docker_tag(version_str: str) -> str:
     """Convert version string to docker tag"""
     return version_str.replace("+", ".")
 
 
-def get_metadata_version(package: str, version_str: str = "") -> str:
+def version_resource() -> str:
+    checks = [get_tag_version, get_build_version, get_pip_version]
+    for i, check in enumerate(checks):
+        if isinstance(check, FunctionType):
+            try:
+                return check()
+            except Exception as e:
+                checks[i] = e
+    for check in checks:
+        if isinstance(check, Exception):
+            log.warning(f"Unable to retrieve version: {check}")
+    raise RuntimeError("Unable to determine version number!")
+
+
+def get_pip_version() -> str:
     """Retrieve the version from the package metadata"""
     try:
-        return str(metadata.version(package))
+        return str(metadata.version("armory-testbed"))
     except metadata.PackageNotFoundError:
-        log.warning(
-            f"ERROR: Unable to find the specified package! Package {package} not installed."
+        raise VersionError(
+            "ERROR: Unable to find Armory package! Armory is not installed."
         )
+
+
+def get_build_version(version_str: str = "") -> str:
+    """Retrieve the version from the build hook"""
+    try:
+        from armory.__about__ import __version__ as version_str
+    except ModuleNotFoundError:
+        raise VersionError("Unable to extract version from __about__.py")
     return version_str
 
 
-def get_tag_version(git_dir: Path = None) -> str:
+def get_tag_version() -> str:
     """Retrieve the version from the most recent git tag"""
     project_root = Path(__file__).parent.parent.parent
     scm_config = {
         "root": project_root,
         "version_scheme": "post-release",
     }
-    if not Path(project_root / ".git").is_dir():
-        log.error("ERROR: Unable to find `.git` directory!")
-        return
-    return setuptools_scm.get_version(**scm_config)
-
-
-def get_build_hook_version(version_str: str = "") -> str:
-    """Retrieve the version from the build hook"""
-    try:
-        from armory.__about__ import __version__ as version_str
-    except ModuleNotFoundError:
-        log.warning("ERROR: Unable to extract version from __about__.py")
-    return version_str
+    if Path(project_root / ".git").is_dir():
```

Comment:
Subtle point here: `scm.get_version` could well succeed even if `.git` is not a directory. The failure is that scm failed, the presence of .git is just a proxy indicator of why. If you think this a likely error, you could test for .git presence after scm failure and add a hint to the log.